esx_service/vmdk_ops.py

```diff
@@ -252,31 +253,64 @@ def getVMDKUuid(vmdk_path):
     except:
         return None
 
+# Create and return the devFS path for a VSAN object UUID.
+# Also returns true if clean up is needed.
+# cleanup_vsan_devfs should be called to clean up before
+# removing the VSAN object.
+def get_vsan_devfs_path(uuid):
+
+    logging.debug("Got volume UUID %s", uuid)
+
+    # Objtool creates a link thats usable to 
+    # read write to vsan object.
+    cmd = "{0} {1}".format(OBJ_TOOL_CMD, uuid)
+    rc, out = RunCommand(cmd)
+    fpath="/vmfs/devices/vsan/{0}".format(uuid)
+    if rc == 0 and os.path.isfile(fpath):
+        return fpath, True
+    logging.error("Failed to create devFS node for %s", uuid)
+    return None, False
+
+# Clean up vsan devfs path
+def cleanup_vsan_devfs_path(devfs_path):
+    try:
+        os.remove(devfs_path)
+        logging.debug("Unlinked %s", devfs_path)
+	return True
+    except OSError as ex:
+        logging.error("Failed to remove backing device %s",
+                      devfs_path, str(ex))
+    return False
+
+# Returns the flat file for a VMDK.
+def get_backing_flat_file(vmdk_path):
+    return vmdk_path.replace(".vmdk", "-flat.vmdk")
 
 # Return a backing file path for given vmdk path or none
-# if a backing can't be found.
-def getVMDKBacking(vmdk_path):
-    flatBacking = vmdk_path.replace(".vmdk", "-flat.vmdk")
+# if a backing can't be found. Returns True if clean up 
+# is needed. Do not forget to call cleanup_backing_device when done.
+def get_backing_device(vmdk_path):
+    flatBacking = get_backing_flat_file(vmdk_path)
     if os.path.isfile(flatBacking):
-        return flatBacking
+        return flatBacking, False
```

Comment:
Same comment as above. We can be conditional on the None and always do cleaunp if flatBacking is valid.